custom_components/ha_washdata/profile_store.py

```
@@ -5848,20 +6118,16 @@ async def assign_profile_to_cycle(
-            # Imported reference recording (separate list, never in usage stats):
-            # relabelling just moves which profile's template it seeds.
-            ref = next(
-                (c for c in self.get_reference_cycles() if c.get("id") == cycle_id),
-                None,
-            )
-            if ref is not None:
-                await self._assign_reference_cycle_profile(ref, profile_name)
-                return
+        found, origin = self.find_stored_cycle(cycle_id)
+        if found is None:
             raise ValueError(f"Cycle {cycle_id} not found")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)